### PR TITLE
Task config

### PR DESCRIPTION
**Pull request by kumare3**

### CODE REVIEW

File: flytekit/annotated/reference_task.py [hunk 1]
```diff
@@ -36,10 +36,10 @@ def get_task_structure(self) -> SdkTask:
 
 def reference_task(
```
Comment (wild-endeavor) [inline flytekit/annotated/reference_task.py:37]: Ugh, I should change this...  i'll add it to the list.  these shouldn't be optional.

**Review by wild-endeavor — APPROVED** [review]
This is so much nicer.

File: flytekit/taskplugins/hive/task.py [hunk 2]
```diff
@@ -10,20 +11,29 @@
 from flytekit.types.schema import FlyteSchema
 
 
-class HiveTask(SQLTask):
+@dataclass
+class HiveConfig(object):
+    cluster_label: str
+    tags: Optional[List[str]] = None
```
Comment (katrogan) [inline flytekit/taskplugins/hive/task.py:17]: Can you add a comment on what these params are for?

**Review by katrogan — APPROVED** [review]

File: flytekit/taskplugins/hive/task.py (hunk 2 above)
Comment (kumare3) [reply flytekit/taskplugins/hive/task.py:17]: doing it now

**Pull request merged**